src/main/python/systemds/operator/nodes/source.py

```diff
@@ -54,10 +53,7 @@ def get_func(self, sds_context, source_name) -> MethodType:
         output_object = self.parse_outputs()
 
         definition = f'def {self._name}(self{argument_string}):'
-        if self._outputs is None:
-            output = f'out = {output_object}(self.sds_context, {operation}, named_input_nodes=named_arguments, output_type=OutputType.NONE)'
-        else:
-            output = f'out = {output_object}(self.sds_context, {operation}, named_input_nodes=named_arguments)'
```

Comment:
the ifelse is not needed anymore since we can infer the output type directly from the output_object